src/UI/Internal/ToggableList.elm

```diff
@@ -0,0 +1,123 @@
+module UI.Internal.ToggableList exposing (Config, view)
+
+import Element exposing (Attribute, Element, fill, px)
+import Element.Background as Background
+import Element.Border as Border
+import Element.Font as Font
+import UI.Icon as Icon
+import UI.Internal.Basics exposing (ifThenElse)
+import UI.Palette as Palette exposing (brightnessDarkest, brightnessLight, brightnessLightest, brightnessMiddle, toneGray, tonePrimary)
+import UI.RenderConfig exposing (RenderConfig)
+import UI.SelectList as SelectList exposing (selectList)
+import UI.Size as Size
+import UI.Text as Text
+import UI.Utils.Element exposing (zeroPadding)
+
+
+type alias Config object msg =
+    { detailsShowLabel : String
+    , detailsCollapseLabel : String
+    , coverView : RenderConfig -> Palette.Color -> object -> Bool -> Element msg
+    , details : object -> List ( String, Element msg )
+    , selectMsg : object -> msg
+    , isSelected : object -> Bool
+    }
+
+
+view : RenderConfig -> Config object msg -> List object -> Element msg
+view renderConfig config items =
+    let
+        itemView parentCfg selected item =
+            if selected then
+                selectedRow parentCfg config item
+
+            else
+                defaultRow parentCfg config selected item
+    in
+    selectList config.selectMsg itemView
+        |> SelectList.withOptions items
+        |> SelectList.withSelected config.isSelected
+        |> SelectList.toEl renderConfig
+
+
+defaultRow : RenderConfig -> Config object msg -> Bool -> object -> Element msg
+defaultRow renderConfig config selected object =
+    Element.row
+        [ Element.width fill
+        , Element.paddingEach { top = 11, bottom = 11, left = 28, right = 12 }
+        , Element.height Element.shrink
+        ]
+        [ config.coverView renderConfig (titleColor selected) object selected
+            |> Element.el
+                [ Element.width fill
+                , Element.centerY
+                , Element.clipX
+                , Element.paddingXY 0 5
+                ]
+        , ifThenElse selected
+            (Icon.toggleUp config.detailsCollapseLabel)
+            (Icon.toggleDown config.detailsShowLabel)
+            |> Icon.withSize Size.large
+            |> Icon.withColor (titleColor selected)
+            |> Icon.toEl renderConfig
+            |> Element.el
+                [ Font.center
+                , Element.width (px 32)
+                , Element.paddingXY 0 6
+                , Element.centerY
+                ]
+        ]
+
+
+selectedRow : RenderConfig -> Config object msg -> object -> Element msg
+selectedRow renderConfig config object =
+    Element.column [ Element.width fill ]
+        [ defaultRow renderConfig config True object
+        , object
+            |> config.details
+            |> List.map (someCard renderConfig)
+            |> Element.column toggleableCard
+        ]
+
+
+someCard : RenderConfig -> ( String, Element msg ) -> Element msg
```

Comment:
I'm not sure about this name, maybe we can have something more explicit? like fieldCard? mobileFieldCard? or something better than my poor suggestions hahah